core/runtime.rs

```diff
@@ -647,6 +647,48 @@ impl JsRuntime {
     scope.perform_microtask_checkpoint();
   }
 
+  /// Waits for the given value to resolve while polling the event loop.
+  ///
+  /// This future resolves when either the value is resolved or the event loop runs to
+  /// completion.
+  pub async fn resolve_value(
+    &mut self,
+    global: v8::Global<v8::Value>,
+  ) -> Result<v8::Global<v8::Value>, AnyError> {
+    poll_fn(|cx| {
```

Comment:
Yeah I find it useful, especially in the context of this PR. We don't use this pattern much, but I can see it being quite popular - executing a script that returns a promise and polling event loop till the promise is resolved.

I'm not sure if there's a point in `resolve_value`; maybe it should be `resolve_promise` that would assert that passed value is a promise?

@ry any comments?